### PR TITLE
Update some canvas properties as enums instead of DOMString

### PR DESCRIPTION
**Pull request by saurvs**
Fixes https://github.com/servo/servo/issues/9617

I'm going to incrementally rollout commits and squash them finally.

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9689)

### CODE REVIEW

**Comment by saurvs** [comment]
@jdm In `components/script/dom/canvasrenderingcontext2d.rs `, the functions `LineCap(&self)` and `SetLineCap(&self, cap_str: DOMString)` return and accept a `DOMString`. Should I modify them to return and accept an appropriate `enum` instead?

**Comment by Ms2ger** [comment]
Yep; I don't expect your build to succeed without that change.

**Comment by saurvs** [comment]
@Ms2ger Does this change make sense?

**Comment by jdm** [comment]
@saurvs There are still build errors, so there's some more work to do. Are you building locally?

**Comment by nox** [comment]
```
components/script/dom/canvasrenderingcontext2d.rs:1212:5: 1214:6 error: method `LineCap` has an incompatible type for trait:
 expected enum `dom::bindings::codegen::Bindings::CanvasRenderingContext2DBinding::CanvasLineCap`,
    found enum `canvas_traits::LineCapStyle` [E0053]
components/script/dom/canvasrenderingcontext2d.rs:1212     fn LineCap(&self) -> LineCapStyle {
components/script/dom/canvasrenderingcontext2d.rs:1213         self.state.borrow().line_cap
components/script/dom/canvasrenderingcontext2d.rs:1214     }
components/script/dom/canvasrenderingcontext2d.rs:1212:5: 1214:6 help: run `rustc --explain E0053` to see a detailed explanation
components/script/dom/canvasrenderingcontext2d.rs:1217:5: 1220:6 error: method `SetLineCap` has an incompatible type for trait:
 expected enum `dom::bindings::codegen::Bindings::CanvasRenderingContext2DBinding::CanvasLineCap`,
    found enum `canvas_traits::LineCapStyle` [E0053]
components/script/dom/canvasrenderingcontext2d.rs:1217     fn SetLineCap(&self, cap: LineCapStyle) {
components/script/dom/canvasrenderingcontext2d.rs:1218         self.state.borrow_mut().line_cap = cap;
components/script/dom/canvasrenderingcontext2d.rs:1219         self.ipc_renderer.send(CanvasMsg::Canvas2d(Canvas2dMsg::SetLineCap(cap))).unwrap();
components/script/dom/canvasrenderingcontext2d.rs:1220     }
components/script/dom/canvasrenderingcontext2d.rs:1217:5: 1220:6 help: run `rustc --explain E0053` to see a detailed explanation
components/script/dom/canvasrenderingcontext2d.rs:1223:5: 1225:6 error: method `LineJoin` has an incompatible type for trait:
 expected enum `dom::bindings::codegen::Bindings::CanvasRenderingContext2DBinding::CanvasLineJoin`,
    found enum `canvas_traits::LineJoinStyle` [E0053]
components/script/dom/canvasrenderingcontext2d.rs:1223     fn LineJoin(&self) -> LineJoinStyle {
components/script/dom/canvasrenderingcontext2d.rs:1224         self.state.borrow().line_join
components/script/dom/canvasrenderingcontext2d.rs:1225     }
components/script/dom/canvasrenderingcontext2d.rs:1223:5: 1225:6 help: run `rustc --explain E0053` to see a detailed explanation
components/script/dom/canvasrenderingcontext2d.rs:1228:5: 1231:6 error: method `SetLineJoin` has an incompatible type for trait:
 expected enum `dom::bindings::codegen::Bindings::CanvasRenderingContext2DBinding::CanvasLineJoin`,
    found enum `canvas_traits::LineJoinStyle` [E0053]
components/script/dom/canvasrenderingcontext2d.rs:1228     fn SetLineJoin(&self, join: LineJoinStyle) {
components/script/dom/canvasrenderingcontext2d.rs:1229         self.state.borrow_mut().line_join = join;
components/script/dom/canvasrenderingcontext2d.rs:1230         self.ipc_renderer.send(CanvasMsg::Canvas2d(Canvas2dMsg::SetLineJoin(join))).unwrap();
components/script/dom/canvasrenderingcontext2d.rs:1231     }
components/script/dom/canvasrenderingcontext2d.rs:1228:5: 1231:6 help: run `rustc --explain E0053` to see a detailed explanation
error: aborting due to 4 previous errors
```

**Comment by saurvs** [comment]
@jdm The build works now.

File: components/script/dom/webidls/CanvasRenderingContext2D.webidl [hunk 1]
```diff
@@ -194,4 +200,3 @@ interface CanvasPathMethods {
   //                             double rotation, double startAngle, double endAngle,
   //                             boolean anticlockwise);
 };
-
```
Comment (KiChjang) [inline components/script/dom/webidls/CanvasRenderingContext2D.webidl]: Undo this change please.

**Comment by jdm** [comment]
@saurvs Thanks! This looks great; I can't think of any way to avoid the conversion between the two enum values, since `script` has to depend on `canvas_msg`. Once you add back the newline, I'll merge this!

**Comment by jdm** [comment]
@bors-servo: retry
* #7791
